iis-common/src/test/java/eu/dnetlib/iis/common/java/jsonworkflownodes/SelectiveTestingConsumer.java

```diff
@@ -0,0 +1,144 @@
+package eu.dnetlib.iis.common.java.jsonworkflownodes;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Properties;
+
+import org.apache.avro.specific.SpecificRecord;
+import org.apache.commons.beanutils.PropertyUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.log4j.Logger;
+
+import com.google.common.base.Preconditions;
+
+import eu.dnetlib.iis.common.OrderedProperties;
+import eu.dnetlib.iis.common.java.PortBindings;
+import eu.dnetlib.iis.common.java.Process;
+import eu.dnetlib.iis.common.java.io.CloseableIterator;
+import eu.dnetlib.iis.common.java.io.DataStore;
+import eu.dnetlib.iis.common.java.io.FileSystemPath;
+import eu.dnetlib.iis.common.java.io.JsonUtils;
+import eu.dnetlib.iis.common.java.porttype.AnyPortType;
+import eu.dnetlib.iis.common.java.porttype.PortType;
+
+/**
+ * Avro datastores selective fields testing consumer. 
+ * 
+ * Requirements are provided as comma separated list of properties files holding field requirements for each individual input object. 
+ * Properties keys are paths to the fields we want to test in given object, properties values are expected values (with special markup for null: $NULL$).
+ * 
+ * Number of input avro records should be equal to the number of provided properties files.
+ *   
+ * @author mhorst
+ */
+public class SelectiveTestingConsumer implements Process {
+
+    private static final String PROPERTIES_CSV = "expectation_properties_csv";
```

Comment:
I wouldn't use the word 'properties', but just expected values or something like this.